App/Sagas/TextileSagas.js

```diff
@@ -239,9 +239,26 @@ export function * shareImage ({thread, hash, caption}) {
 
 export function * photosTask () {
   try {
-    const photos = yield call(queryPhotos)
+    const camera = yield select(TextileSelectors.camera)
+    const processed = camera && camera.processed ? camera.processed : []
+
+    let allPhotos = yield call(getAllPhotos)
+    if (camera === undefined || camera.processed === undefined) {
+      for (const uri of allPhotos.map((p) => p.uri)) {
+        yield put(TextileActions.newImage(uri))
+      }
+      allPhotos = []
+    }
+
+    const newPhotos = allPhotos.filter((photo) => {
+      return processed.indexOf(photo.uri) === -1
+    })
+
+    const photos = yield call(scalePhotos, newPhotos)
+
     for (const photo of photos.reverse()) {
       const multipartData = yield call(IPFS.addImageAtPath, photo.path, photo.thumbPath, 'default')
+      yield put(TextileActions.newImage(photo.uri))
```

Comment:
Wondering if we need this new redux action `newImage` or can we just add a `uri` property to the `imageAdded` action that is triggered 3 lines below?

Reply:
okay, was putting it up here thinking through any errors we might hit below it wouldn't keep retrying. but happy to do that

Reply:
Yea good call. Seems like a good idea, but that alone won't save off any error state for an individual photo because the `catch` we're in here is outside this `for` loop and just dispatches a general `photosTaskError`. To complete what you started here, there should be an inner (within the `for` loop) `try/catch` that catches errors for processing an individual photo, and dispatched a new action, `photoProcessingError(id, error)`, then the reducer can update the photo state with that error. 

If you want to add that here, cool, or we can save it for an additional PR.